### PR TITLE
Add iOS Banner monitoring

### PR DESCRIPTION
**Pull request by ddramowicz**
I've added both Braze and Olgil monitoring for banners but since the Braze banner is not implemented yet I left its threshold at 0 and did not add it to the `allFeaturesWithMonitoring` list.
The `olgil_banner` threshold number is yesterday's impressions on all app versions, halved.
I tested the Olgil one locally using a fake minimum threshold (a much too large one) and got the alert in the Anghammarad google group.
I'm not sure if adding the Braze one now as I've done is a good idea or not, so I could use some feedback on that.

### CODE REVIEW

File: src/main/scala/com/gu/datalakealerts/Features.scala [hunk 1]
```diff
@@ -138,4 +138,61 @@ object Features {
...
+
+      }
+    }
+  }
+
+  case object BrazeBanner extends Feature {
+    override val id = "braze_banner"
+
+    override def monitoringQueryResult(resultSet: ResultSet, minimumImpressionsThreshold: Int): MonitoringQueryResult = {
+      val impressionCountsByAppVersion = ImpressionCounts.getImpressionCounts(resultSet)
```
Comment (jacobwinch) [inline src/main/scala/com/gu/datalakealerts/Features.scala]: Although this isn't a new problem introduced by this PR, this code is now duplicated several times. We should think about how this code can be shared between features (probably best for that to be addressed in a separate PR though)

File: src/main/scala/com/gu/datalakealerts/Features.scala [hunk 2]
```diff
@@ -138,4 +138,61 @@ object Features {
       }
     }
   }
+  case object OlgilBanner extends Feature {
```
Comment (jacobwinch) [inline src/main/scala/com/gu/datalakealerts/Features.scala]: This [line](https://github.com/guardian/data-lake-alerts/pull/32/files#diff-ac7fae1faa0a8cd922c4949a768416b5R25) means that [we will automatically schedule monitoring for both platforms unless otherwise specified](https://github.com/guardian/data-lake-alerts/blob/master/src/main/scala/com/gu/datalakealerts/SchedulerLambda.scala#L29-L31). 

Given that attempting to monitor banner impressions on Android is going to throw an exception, we should add this line to both features:

```
val platformsToMonitor = List(iOS)
```

**Comment by jacobwinch** [comment]
> I'm not sure if adding the Braze one now as I've done is a good idea or not, so I could use some feedback on that.

I think it's sensible to do this (as long as we create a card to remind ourselves to finalise this later on).

File: src/main/scala/com/gu/datalakealerts/Features.scala (hunk 2 above)
Comment (jacobwinch) [reply src/main/scala/com/gu/datalakealerts/Features.scala]: Once you've added this line you should be able to confirm that Android Olgil/Braze Banner monitoring is not being scheduled anymore by running: https://github.com/guardian/data-lake-alerts#monitoring-schedule

**Comment by ddramowicz** [comment]
> > I'm not sure if adding the Braze one now as I've done is a good idea or not, so I could use some feedback on that.
> 
> I think it's sensible to do this (as long as we create a card to remind ourselves to finalise this later on).

I've created this ticket in Jira to do this: https://theguardian.atlassian.net/browse/GLA-1579
